commit 61dfaa4d96e5657bd7470f58763ebc9df432f745
Author: Seth Forshee <seth.forshee@canonical.com>
Date:   Fri Aug 25 14:47:29 2017 -0500

    UBUNTU: SAUCE: selftests/powerpc: Use snprintf to construct DSCR sysfs interface paths
    
    BugLink: http://bugs.launchpad.net/bugs/1710904
    
    Currently sprintf is used, and while paths should never exceed
    the size of the buffer it is theoretically possible since
    dirent.d_name is 256 bytes. As a result this trips
    -Wformat-overflow, and since the test is built with -Wall -Werror
    the causes the build to fail. Switch to using snprintf and skip
    any paths which are too long for the filename buffer.
    
    Signed-off-by: Seth Forshee <seth.forshee@canonical.com>

diff --git a/tools/testing/selftests/powerpc/dscr/dscr_sysfs_test.c b/tools/testing/selftests/powerpc/dscr/dscr_sysfs_test.c
index 17fb1b43c320..1899bd85121f 100644
--- a/tools/testing/selftests/powerpc/dscr/dscr_sysfs_test.c
+++ b/tools/testing/selftests/powerpc/dscr/dscr_sysfs_test.c
@@ -53,6 +53,8 @@ static int check_all_cpu_dscr_defaults(unsigned long val)
 	}
 
 	while ((dp = readdir(sysfs))) {
+		int len;
+
 		if (!(dp->d_type & DT_DIR))
 			continue;
 		if (!strcmp(dp->d_name, "cpuidle"))
@@ -60,7 +62,9 @@ static int check_all_cpu_dscr_defaults(unsigned long val)
 		if (!strstr(dp->d_name, "cpu"))
 			continue;
 
-		sprintf(file, "%s%s/dscr", CPU_PATH, dp->d_name);
+		len = snprintf(file, LEN_MAX, "%s%s/dscr", CPU_PATH, dp->d_name);
+		if (len >= LEN_MAX)
+			continue;
 		if (access(file, F_OK))
 			continue;
 
